cfme/configure/access_control.py

```diff
@@ -1444,32 +1451,13 @@ def step(self):
 @navigator.register(Tenant, 'Details')
 class TenantDetails(CFMENavigateStep):
     VIEW = DetailsTenantView
-    prerequisite = NavigateToSibling('All')
+    prerequisite = NavigateToAttribute('collection', 'All')
 
     def step(self):
         self.prerequisite_view.accordions.accesscontrol.tree.click_path(
             self.obj.appliance.server_region_string(), 'Tenants', *self.obj.tree_path)
 
 
-@navigator.register(Tenant, 'Add')
```

Comment:
**REQUIRED:** Where did this go? Is it already on ```TenantCollection```?

Reply:
To add a new Tenant/Project, we first need to navigate to the parent, under which we want to add it. 
And as per `Collection` design `create` method should be on `TenantCollection`. 

So writing `prerequisite = NavigateToAttribute(Tenant, 'Detail')` in `Add` navigator registered with collection would need the `Tenant` object which we want to navigate to. And passing args to navigation didn't seem good design (and would probably yell if done?) 

So I've removed the `Add` and doing the some steps of navigation inside `create` method on collection. Its just like navigating first to `Details` and then going to `Add` page.               
`create` func [code](https://github.com/vatsalparekh/integration_tests/blob/1b394e02e25c23176f6806d43948df50e22c1b9f/cfme/configure/access_control.py#L1305)

Reply:
The prerequisite for adding a tenant should be the 'All' view registered against TenantCollection, not the details view.

Reply:
@mshriver but then that `Add` view need to be registered with `Tenant` to have `parent_tenant` accessible, and we're calling it from Collection so kind of contradictory in design
@psav help please in deciding the final change

Reply:
So, within `TenantCollection.create()` can we not do:

```python
# pseudocode
tenant_entity = self.instantiate(**kwargs)
view=navigate_to(tenant_entity, 'add')
view.fill
return tenant_entity

Reply:
**REQUIRED:** This needs to be responded to

Reply:
@mshriver 
1) The functionality to add a tenant can be accessed from the `Detail` view of a `tenant`, so it can't be registered with `All` as prerequisite
2) with the pseudocode given, issue is in prerequisite of `add` view; how would we navigate to `details` when we don't have the instantiated `tenant_entity` accessible inside the view?

Reply:
I understand now, thanks